go/geth_adapter/adapter.go

```diff
@@ -230,13 +194,215 @@ func (a *gethInterpreterAdapter) Run(contract *geth.Contract, input []byte, read
 	return res.Output, nil
 }
 
+func debugRunStart(input []byte, readOnly bool) {
+	fmt.Printf("Begin of interpreter:\n")
+	fmt.Printf("\tInput:  %v\n", input)
+	fmt.Printf("\tStatic: %v\n", readOnly)
+}
+
+func debugRunEnd(res tosca.Result) {
+	fmt.Printf("End of interpreter:\n")
+	fmt.Printf("\tSuccess:  %v\n", res.Success)
+	fmt.Printf("\tOutput:   %v\n", res.Output)
+	fmt.Printf("\tGas Left: %v\n", res.GasLeft)
+	fmt.Printf("\tRefund:   %v\n", res.GasRefund)
+}
+
+func shiftRefundBack(stateDB geth.StateDB, err error, refundShift uint64) {
```

Comment:
From the name of this function, one would not expect that it also pays back the refund. One would expect that this function returns the corrected refund.

How about renaming it to `undoRefundShift`?